### PR TITLE
added z_rotate option

### PR DESCRIPTION
**Pull request by csch0**
I always print (at least big part) with 45 degree rotated on z axes. Reason is that the infill is a x or y axis move only, made this available as an option to SuperSlicer, not sure if that is the best location of the setting. So as usual open to discussion.

image

### CODE REVIEW

**Comment by kelchm** [comment]
I think this is a great concept -- definitely something I would like to see supported in a more direct way in SuperSlier.

To understand where in the hierarchy something like this should live, I think we have to consider the motivations for _why_ users might want to print parts at a 45º angle. Here's a (likely incomplete) list:

1. To match printer kinematics (EX: 45ª moves require only a single axis on CoreXY)
2. To more reliably control seam positioning (IE: combine 45º rotation with `Rear` seam position)

Are there other use cases that I'm overlooking?

**Comment by supermerill** [comment]
@kelchm convinced me that it needs to be on the printer side.

I've merged your first commit.
Ideally, when changing the printer, we could transfer the rotation from the previous to the next printer (so reverting the rotation, then applying the new one). But I don't have time to tinker to find the right place to do that right now.

**Comment by supermerill** [comment]
I tested it, and it's a bit limited: it doesn't work on importing 3mf (only the object) and don't rotate the built-in cube.
It needs some improvement before the release

**Comment by csch0** [comment]
Will have a look tomorrow then.

**Comment by JRHeaton** [comment]
I like this idea because I manually rotate my objects 45 degrees on cartesian machines as well. This being so that, perhaps conversely to you, I am utilizing both X and Y axes in movement and reducing the distance and necessary accel for each. On a CoreXY, kinematics are different and thus you see batch or speed prints with things aligned to the bed edges (normal orientation) so that both motors are used.

TL;DR: Cartesian: 45 degree, CoreXY 0 degree rotation for optimal speed. 

If you're wanting (for some reason) to make single-motor-only moves for infill, as has been stated there is an option to rotate/set the infill angle already and rotating the model itself does not rotate the infill. Perhaps I misunderstand the intentions from the wording "Reason is that the infill is a x or y axis move only"

**Comment by supermerill** [comment]
@JRHeaton look at printer -> general -> size -> preferred orientation
note that it only works when importing stl. Not projects or add shape.

**Comment by JRHeaton** [comment]
@supermerill I'm confused then, why would we need an additional option for Z rotate? That existing setting seems to do exactly what is desired, no?

**Comment by supermerill** [comment]
?
this setting is this pull request merged.
I just forgot to close this pr.

**Comment by JRHeaton** [comment]
>  I just forgot to close this pr.

This was 75 days ago. Can you close this PR now?
